unit_scaling/utils.py

```diff
@@ -222,7 +222,7 @@ def trace(
         """Adds the `target_to_function` dict to the graph so the interpreter can use it
         downstream."""
         graph = super().trace(root, concrete_args)
-        if graph._tracer_extras is None:
+        if not hasattr(graph, "_tracer_extras") or graph._tracer_extras is None:
```

Comment:
If I understand the test (precendence of `or` and `not` always gets me), could this use `getattr(graph, "_tracer_extras", None) is None`?